crypto/dilithium/pqcrystals_dilithium_ref_common/poly.c

```diff
@@ -505,12 +505,11 @@ void poly_challenge(ml_dsa_params *params, poly *c, const uint8_t *seed) {
   unsigned int i, b, pos;
   uint64_t signs;
   uint8_t buf[SHAKE256_RATE];
-  keccak_state state;
+  KECCAK1600_CTX state;
 
-  shake256_init(&state);
-  shake256_absorb(&state, seed, params->c_tilde_bytes);
-  shake256_finalize(&state);
-  shake256_squeezeblocks(buf, 1, &state);
+  SHAKE_Init(&state, SHAKE256_BLOCKSIZE);
+  SHA3_Update(&state, seed, params->c_tilde_bytes);
+  dilithium_shake256_squeeze(&state, buf, 1);
```

Comment:
why can't we just use `SHA_Final` directly?

Reply:
actually, why can't we use `SHAKE256` instead of the sequence of Init, Update, Final?

Reply:
Because we need an additional squeeze down at line 524 below. If we call `SHAKE256` we cannot additionally squeeze output later.

Reply:
However, I agree that we can use `SHA_Final` directly, and have modified the PR to do this for both SHAKE256 and SHAKE 128 in [f4cc0fe](https://github.com/aws/aws-lc/pull/2001/commits/f4cc0fe839ef068d1b60afaf80ab0a2f55ca91a9)